node/activation_service.go

```diff
@@ -0,0 +1,712 @@
+			app.Config.DataDir(),
+			err,
+		)
+	}
+
+	/* Setup monitoring */
+	app.errCh = make(chan error, 100)
+	if app.Config.PprofHTTPServer {
+		logger.With().Info("starting pprof server", log.String("address", app.Config.PprofHTTPServerListener))
+		app.pprofService = &http.Server{Addr: app.Config.PprofHTTPServerListener}
+		app.eg.Go(func() error {
+			if err := app.pprofService.ListenAndServe(); err != nil {
+				app.errCh <- fmt.Errorf("cannot start pprof http server: %w", err)
+			}
+			return nil
+		})
+		if app.Config.PprofMutexProfile {
+			// this will set the mutex profiling to sample a third of all lock events
+			runtime.SetMutexProfileFraction(3)
+		}
+		if app.Config.PprofBlockProfile {
+			// record block sample for every block event that takes more than 10 milliseconds
+			runtime.SetBlockProfileRate(int(10 * time.Millisecond))
+		}
+	}
+
+	if app.Config.ProfilerURL != "" {
+		app.profilerService, err = pyroscope.Start(pyroscope.Config{
+			ApplicationName: app.Config.ProfilerName,
+			// app.Config.ProfilerURL should be the pyroscope server address
+			// TODO: AuthToken? no need right now since server isn't public
+			ServerAddress: app.Config.ProfilerURL,
+			// by default all profilers are enabled,
+		})
+		if err != nil {
+			return fmt.Errorf("cannot start profiling client: %w", err)
+		}
+	}
+
+	var preserved *checkpoint.PreservedData
+	if app.Config.Recovery.Uri != "" {
+		preserved, err = app.loadCheckpoint(ctx)
+		if err != nil {
+			return fmt.Errorf("loading checkpoint: %w", err)
+		}
+	}
+
+	/* Initialize all protocol services */
+	app.clock, err = timesync.NewClock(
+		timesync.WithLayerDuration(app.Config.LayerDuration),
+		timesync.WithTickInterval(1*time.Second),
+		timesync.WithGenesisTime(app.Config.Genesis.GenesisTime.Time()),
+		timesync.WithLogger(app.addLogger(ClockLogger, logger).Zap()),
+	)
+	if err != nil {
+		return fmt.Errorf("cannot create clock: %w", err)
+	}
+
+	if err := app.setupDBs(ctx, logger); err != nil {
+		return err
+	}
+
+	if err := app.initActivationServiceServices(ctx); err != nil {
+		return fmt.Errorf("init services: %w", err)
+	}
+
+	if app.Config.CollectMetrics {
+		metrics.StartMetricsServer(app.Config.MetricsPort)
+	}
+
+	if app.Config.PublicMetrics.MetricsURL != "" {
+		id := hash.Sum([]byte(app.host.ID()))
+		metrics.StartPushingMetrics(
+			app.Config.PublicMetrics.MetricsURL,
+			app.Config.PublicMetrics.MetricsPushUser,
+			app.Config.PublicMetrics.MetricsPushPass,
+			app.Config.PublicMetrics.MetricsPushHeader,
+			app.Config.PublicMetrics.MetricsPushPeriod,
+			types.Hash32(id).ShortString(), app.Config.Genesis.GenesisID().ShortString())
+	}
+
+	if err := app.startActivationServiceServices(ctx); err != nil {
+		return fmt.Errorf("start services: %w", err)
+	}
+
+	// need post verifying service to start first
+	if preserved != nil {
+		app.preserveAfterRecovery(ctx, *preserved)
+	} else {
+		app.log.Info("no need to preserve data after recovery")
+	}
```

Comment:
This is probably not needed on the smashing-service (it's related to checkpoint which should only matter on the node-service).
```suggestion
```

Reply:
Removed with connected to that code.